dnfdaemon-client/commands/repolist/repolist.cpp

```diff
@@ -0,0 +1,189 @@
+        ctx.arg_parser.add_init_value(std::unique_ptr<libdnf::OptionEnum<std::string>>(
+            new libdnf::OptionEnum<std::string>("enabled", {"all", "enabled", "disabled"}))));
+
+    auto all = ctx.arg_parser.add_new_named_arg("all");
+    all->set_long_name("all");
+    all->set_short_description("show all repos");
+    all->set_const_value("all");
+    all->link_value(enable_disable_option);
+
+    auto enabled = ctx.arg_parser.add_new_named_arg("enabled");
+    enabled->set_long_name("enabled");
+    enabled->set_short_description("show enabled repos (default)");
+    enabled->set_const_value("enabled");
+    enabled->link_value(enable_disable_option);
+
+    auto disabled = ctx.arg_parser.add_new_named_arg("disabled");
+    disabled->set_long_name("disabled");
+    disabled->set_short_description("show disabled repos");
+    disabled->set_const_value("disabled");
+    disabled->link_value(enable_disable_option);
+
+    patterns_to_show_options = ctx.arg_parser.add_new_values();
+    auto repos = ctx.arg_parser.add_new_positional_arg(
+        "repos_to_show",
+        libdnf::cli::ArgumentParser::PositionalArg::UNLIMITED,
+        ctx.arg_parser.add_init_value(std::unique_ptr<libdnf::Option>(new libdnf::OptionString(nullptr))),
+        patterns_to_show_options);
+    repos->set_short_description("List of repos to show");
+
+    auto conflict_args =
+        ctx.arg_parser.add_conflict_args_group(std::unique_ptr<std::vector<libdnf::cli::ArgumentParser::Argument *>>(
+            new std::vector<libdnf::cli::ArgumentParser::Argument *>{all, enabled, disabled}));
+
+    all->set_conflict_arguments(conflict_args);
+    enabled->set_conflict_arguments(conflict_args);
+    disabled->set_conflict_arguments(conflict_args);
+
+    auto repolist = ctx.arg_parser.add_new_command(command);
+    repolist->set_short_description("display the configured software repositories");
+    repolist->set_description("");
+    repolist->set_named_args_help_header("Optional arguments:");
+    repolist->set_positional_args_help_header("Positional arguments:");
+    repolist->set_parse_hook_func([this, &ctx](
+                                      [[maybe_unused]] libdnf::cli::ArgumentParser::Argument * arg,
+                                      [[maybe_unused]] const char * option,
+                                      [[maybe_unused]] int argc,
+                                      [[maybe_unused]] const char * const argv[]) {
+        ctx.select_command(this);
+        return true;
+    });
+
+    repolist->register_named_arg(all);
+    repolist->register_named_arg(enabled);
+    repolist->register_named_arg(disabled);
+    repolist->register_positional_arg(repos);
+
+    ctx.arg_parser.get_root_command()->register_command(repolist);
+}
+
+class RepoDbus {
+public:
+    RepoDbus(dnfdaemon::KeyValueMap & rawdata) : rawdata(rawdata){};
+    std::string get_id() { return rawdata["id"]; }
```

Comment:
The method will add item (with default value) to the map if it does not exist? Is it intended? What about to make the methods `const`?

Examples:
1. Will return default value if key does not exist (but value is not added to the map):
`std::string get_id() const { auto it = rawdata.find("name"); return it != rawdata.end() ? it->second : sdbus::Variant(); }`

2. Will generate exception if key does not exist:
`std::string get_id() const { return rawdata.at("id"); }`